DeviceConnectionAndReliabilityReadme.md

```diff
@@ -0,0 +1,62 @@
+# Azure IoT Device Client C# SDK
+
+## Device connection and messaging reliability
+
+### Overview
+
+In this document you will find information about
+
+- The connection authentication and renewal methods.
+- The reconnection logic and retry policies.
+- The timeout controls.
+
+### Connection authentication
+
+Authentication can be done using either
+
+- [SAS tokens](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-security#security-tokens) - [Sample usage](https://github.com/Azure-Samples/azure-iot-samples-csharp/blob/master/iot-hub/Samples/device/TwinSample/Program.cs#L37)
+- [x509 certificates](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-security#supported-x509-certificates) - [Sample usage](https://github.com/Azure-Samples/azure-iot-samples-csharp/blob/master/iot-hub/Samples/device/X509DeviceCertWithChainSample/Program.cs#L43)
+- [Device Provisioning Service](https://docs.microsoft.com/en-us/azure/iot-dps) - [Sample usage](https://github.com/Azure-Samples/azure-iot-samples-csharp/blob/master/iot-hub/Samples/device/PnpDeviceSamples/Thermostat/Program.cs#L90)
+
+When using SAS tokens, authentication can be done by:
+
+- Providing the shared access key and letting the SDK create the SAS tokens by using one of the [CreateFromConnectionString](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/src/DeviceClient.cs#L121) options on the DeviceClient or an authenticationMethod other than DeviceAuthenticationWithTokenRefresh.
```

Comment:
how about we list the auth methods that they can use, instead of saying ..."or an authenticationMethod other than DeviceAuthenticationWithTokenRefresh.". That way we don't leave this ambiguous.

Reply:
I can add a section on the auth methods we provide implementation for. However, customers can have their own implementation as well so I did not say, these are the only ones you can use. What do you think?

Reply:
Language ambiguity about whether the user or the SDK using `CreateFromConnectionString` or `DeviceAuthenticationWithTokenRefresh`.

Also, I don't know if `authenticationMethod` is meant to be a parameter. I changed it to plain text. If it is, then use backticks to render it as code.

```suggestion
- Providing the shared access key and letting the SDK create the SAS tokens. To do this use one of the [CreateFromConnectionString](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/src/DeviceClient.cs#L121) options on the `DeviceClient` or an authentication method other than DeviceAuthenticationWithTokenRefresh.
```

Reply:
Made updates to the language and used back ticks to clarify.